pkg/protocol/engine/commitmentfilter/accountsfilter/commitmentfilter.go

```diff
@@ -155,12 +155,23 @@ func (c *CommitmentFilter) evaluateBlock(block *blocks.Block) {
 	switch signature := block.ProtocolBlock().Signature.(type) {
 	case *iotago.Ed25519Signature:
 		if !accountData.BlockIssuerKeys.Has(iotago.BlockIssuerKeyEd25519FromPublicKey(signature.PublicKey)) {
-			c.events.BlockFiltered.Trigger(&commitmentfilter.BlockFilteredEvent{
-				Block:  block,
-				Reason: ierrors.Wrapf(ErrInvalidSignature, "block issuer account %s does not have public key %s in slot %d", block.ProtocolBlock().IssuerID, signature.PublicKey, block.ProtocolBlock().SlotCommitmentID.Index()),
+			// if the block issuer does not have the public key in the slot commitment, check if it has an implicit account with the corresponding address
+			var hasAddress = false
+			accountData.BlockIssuerKeys.Range(func(element iotago.BlockIssuerKey) {
```

Comment:
Can't we stop the loop if we found the address? maybe we should add a return bool parameter to the `Range` function to abort it?